### ISSUE TITLE
Trick to load language file in module definition page: acceptable?

### ISSUE THREAD

**Issue by st-anton** [Joomla-Bible-Study/Joomla-Bible-Study#404]
Both plugin and module have no code in the backend, but xml based file for parameter settings only. So it is not possible to use a common language file for component and module (backend) or plugin (backend). This is not that bad for plugins, because there are quite few settings; also this settings are very specific.
It is worse for the module: A lot of layout and 'template' settings are done in the backend. So I played around and found a way: Add a dummy field that is not visible and load the language files in the according field definition. See code below.
What's your opinion: Is this acceptable?

mod_biblestudy.xml:
```
 <?xml version="1.0" encoding="utf-8"?>
 <extension type="module" version="2.5" method="upgrade" client="site">
    <name>mod_biblestudy</name>
 ...shortened
    <config>
        <fields name="params">
            <fieldset name="basic"
                      addfieldpath="/administrator/components/com_biblestudy/models/fields">
new:           <field name="dummy" type="loadlanguagefile" default=""/>
                <field name="t" type="sql" description="JBS_MNU_SELECT_TEMPLATE_DESC"
                       label="JBS_MNU_SELECT_TEMPLATE" default="1"
                       query="SELECT id, title FROM #__bsms_templates WHERE published = 1 ORDER BY text ASC"
                       key_field="id" value_field="title" required="true"/>
```
New file loadlanguagefile.php:
```
 class JFormFieldLoadLanguageFile extends JFormField
 {
	protected $type = 'LoadLanguageFile';
	protected $hidden = true;

	public function getLabel() {
        // return an empty string; nothing to display
		return '';
	}
	protected function getInput()
	{
		// get language file; english language as fallback
		$language = JFactory::getLanguage();
		$language->load('com_biblestudy', JPATH_ADMINISTRATOR . '/components/com_biblestudy', 'en-GB', true);
		$language->load('com_biblestudy', JPATH_ADMINISTRATOR . '/components/com_biblestudy', null, true);

        // return an empty string; nothing to display
		return '';
	}
}
```

**Comment by st-anton** [comment]
I continued working with this. It confirmed to me, that this is a resonable way to use and maintain one common language file for component, module and plugin. Only the .sys.ini files will stay.
Unifying the modules language file and the components language file is a effort. Even bigger is the effort to maintain both!
If you share my opinion, please let me know. Then I will merge all files asap.

**Comment by bcordis** [comment]
Let's merge. Please see commit to #399 about the that we may need to move to the language file in the core folder now. What you think?

**Comment by st-anton** [comment]
Did I get it right:
The bootstrap code has to do with the spanning of columns in responsive design. It has nothing to do with language files.
The language file issue appears with the post install messages. <- If this is true, I have to look deeper into that.

**Comment by bcordis** [comment]
Correct post install. It only looks for the file inside the root folders and not in the component.

**Comment by st-anton** [comment]
Is it com_biblestudy/admin/models/migration.php  line 647 ? Is there other code as well, that puts out post install text?

**Comment by st-anton** [comment]
Well, I want to merge my changes into current development, afterwards we will have a look into the post install messages.

**Comment by bcordis** [comment]
That is just the installer. Under 9.0.0.php is a template one.

**Comment by st-anton** [comment]
Can you please give me a hint how I can test the post install? Do I need to have an old database content, e.g. 8.0.0? Is the post installer invoked every time during installation or for update of the DB only?

**Comment by bcordis** [comment]
I have not put any post install on clean installs. Right now it works on migration or updates. I have made one for 9.0.0

**Comment by st-anton** [comment]
As written in #438 (and #399) the postinstall messages cannot be used currently to put a message. Either the postinstall comonent needs to be hacked or the language file needs to be placed in the CORE language folder.

**Comment by bcordis** [comment]
@st-anton so can we close this as this should be fix?

**Comment by st-anton** [comment]
Yes, this can be closed. (NB: I need to fix #480 asap)
